### PR TITLE
updates to offer vertical rhythm

### PR DESCRIPTION
**Pull request by matthewcarleton**
## Description

I've created a vertical rhythm feature. It applies margins to all type level tags and font-sizes to all headings when wrapped in the class .verticalRhythm.

Please review code and also how I did the handlebars file (first time). 

Thanks!

### CODE REVIEW

File: source/_patterns/templates/typography/typography.hbs [hunk 1]
```diff
@@ -0,0 +1,115 @@
+<div class="container-fluid">
+    <div class="pf-verticalRhythm">
```
Comment (andresgalante) [inline source/_patterns/templates/typography/typography.hbs]: We don't use came classes for class names:
https://github.com/patternfly/patternfly-css/blob/master/CODE-GUIDELINES.md#naming-selectors

Can you change it to `pf-vertical-rhythm` please?

File: source/_patterns/templates/typography/vertical-rhythm.hbs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<section class="container-fluid pf-verticalRhythm">
```
Comment (andresgalante) [inline source/_patterns/templates/typography/vertical-rhythm.hbs]: A container or any layout classes should always be on their own, let's create a new div wrapper around the text you'd like to get vertical rythm.

File: source/_patterns/templates/typography/vertical-rhythm.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+title: Vertical Rhythm
+---
+## Overview
+Vertical Rhythm offers a vertical flow for pages that have content that live outside of components. When the class .verticalRhythm is applied all typography level tags will receive a bottom margin. All headings will also be given the appropriate size and weight.
+
+## Usage
+
+| Class | Usage |
+| -- | -- |
+| `.pf-verticalRhythm` **Applied to:** wrapping `<div>`|  **Outcome:** applies top and bottom margins to all headings, paragraphs and lists. **Required:** No
```
Comment (andresgalante) [inline source/_patterns/templates/typography/vertical-rhythm.md]: **Applied to:** `<div>`, `<section>` or `<article>`. 
what else can it be?
**Required:** Yes
It is required if you want to add vertical rhythm. The answer no is usually for modifiers.

File: source/_patterns/templates/typography/vertical-rhythm.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+---
+title: Vertical Rhythm
+---
+## Overview
+Vertical Rhythm offers a vertical flow for pages that have content that live outside of components. When the class .verticalRhythm is applied all typography level tags will receive a bottom margin. All headings will also be given the appropriate size and weight.
```
Comment (andresgalante) [inline source/_patterns/templates/typography/vertical-rhythm.md]: Remember to change the class name here too

File: source/scss/_pf-reboot.scss [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//
+// Typography
+// --------------------------------------------------
+ol,ul,dl{
```
Comment (andresgalante) [inline source/scss/_pf-reboot.scss]: Space after comma and before {

File: source/scss/_pf-reboot.scss [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  margin-bottom:0;
+}
+p{
+  margin-bottom: 0;
```
Comment (andresgalante) [inline source/scss/_pf-reboot.scss]: I would not remove this globally, sometimes we need to use paragraph inside a component, like in cards.

File: source/scss/_pf-reboot.scss [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//
+// Typography
+// --------------------------------------------------
+ol,ul,dl{
+  margin-bottom:0;
```
Comment (andresgalante) [inline source/scss/_pf-reboot.scss]: I don't know if we should remove it on lists globally, maybe yes.

File: source/scss/_pf-type.scss [hunk 8]
```diff
@@ -23,11 +23,68 @@ html {  font-size: $pf-font-size-root;}
 
 body {  font-weight: $pf-font-weight-regular;}
 
-h4, .h4 { font-weight: $pf-font-weight-semi-bold; }
+strong { font-weight: $pf-font-weight-bold; }
 
-strong{ font-weight:   $pf-font-weight-bold; }
+p { margin-bottom: $pf-typography-margin-bottom; }
```
Comment (andresgalante) [inline source/scss/_pf-type.scss]: If it's added here, why would we remove it on reboot?

File: source/scss/_pf-type.scss [hunk 9]
```diff
@@ -23,11 +23,68 @@ html {  font-size: $pf-font-size-root;}
...
 
-h1, h2, h3, h4, h5, h6 {
-  margin-top: $pf-headings-margin-top;
+h1{
```
Comment (andresgalante) [inline source/scss/_pf-type.scss]: spaces before the { 😄

File: source/scss/_pf-type.scss [hunk 10]
```diff
@@ -23,11 +23,68 @@ html {  font-size: $pf-font-size-root;}
...
+      $pf-headings-margin-top,
+      $pf-typography-margin-bottom
+      )
+}
+
+p,ul,ol,dl,blockquote,pre,td,th,textarea{
+  .pf-verticalRhythm &{
+    margin-bottom: $pf-typography-margin-bottom;
+    &:last-child{
+      margin-bottom:0; // last item in verticalRhythm parent has no margin-bottom
```
Comment (andresgalante) [inline source/scss/_pf-type.scss]: space after the `:` we SO need a linter!!

File: source/scss/_pf-type.scss [hunk 11]
```diff
@@ -23,11 +23,68 @@ html {  font-size: $pf-font-size-root;}
...
+h3{
+  @include pf-typography-styles();
+}
+
+h4{
+  @include pf-typography-styles($pf-font-weight-semi-bold);
+}
+
+.h1, .pf-verticalRhythm h1{
+      @include pf-typography-styles(
```
Comment (andresgalante) [inline source/scss/_pf-type.scss]: 2 space indentation. I am the human linter

File: source/_patterns/templates/typography/typography.hbs [hunk 12]
```diff
@@ -0,0 +1,115 @@
+<div class="container-fluid">
+    <div class="pf-vertical-rhythm">
```
Comment (andresgalante) [inline source/_patterns/templates/typography/typography.hbs]: This should be inside `col-md-12`

File: source/_patterns/templates/typography/vertical-rhythm.hbs [hunk 13]
```diff
@@ -0,0 +1,34 @@
+<section class="container-fluid">
+  <div class="pf-vertical-rhythm">
```
Comment (andresgalante) [inline source/_patterns/templates/typography/vertical-rhythm.hbs]: same here

File: source/scss/_pf-mixins.scss [hunk 14]
```diff
@@ -102,3 +102,20 @@ limitations under the License.
...
+// Used in conjunction with .vertical-rhythm to apply size, weight, and margins to type.
+
+@mixin pf-typography-styles(
+  $weight: $pf-font-weight-regular,
+  $size: $font-size-base,
+  $margin-top: null,
+  $margin-bottom: null
+  )
+  {
+  font-size:$size;
```
Comment (andresgalante) [inline source/scss/_pf-mixins.scss]: human linter!

**Comment by matthewcarleton** [comment]
updated, package-lock.json should be gone now.

**Comment by andresgalante** [comment]
👍  thanks a lot @matthewcarleton!

File: source/_patterns/templates/typography/vertical-rhythm.md [hunk 15]
```diff
@@ -0,0 +1,11 @@
...
+title: Vertical Rhythm
+---
+## Overview
+Vertical Rhythm offers a vertical flow for pages that have content that live outside of components. When the class .pf-vertical-rhythm is applied all typography level tags will receive a margin. All headings will also be given the appropriate size and weight.
+
+## Usage
+
+| Class | Usage |
+| -- | -- |
+| `.pf-vertical-rhythm` **Applied to:** wrapping `<div>, <section> or <article>`|  **Outcome:** applies top and bottom margins to all headings, paragraphs and lists. **Required:** Yes
```
Comment (andresgalante) [inline source/_patterns/templates/typography/vertical-rhythm.md]: ... and font size and weight to headings

File: source/_patterns/templates/typography/vertical-rhythm.hbs [hunk 16]
```diff
@@ -0,0 +1,34 @@
+<section class="container-fluid">
+    <div class="row">
+      <div class="col-md-12">
+        <div class="pf-vertical-rhythm">
+          <h1>Vertical Ryhthm</h1>
```
Comment (srambach) [inline source/_patterns/templates/typography/vertical-rhythm.hbs]: Typo :) "Rhythm"

File: source/_patterns/templates/typography/vertical-rhythm.md (hunk 15 above)
Comment (srambach) [inline source/_patterns/templates/typography/vertical-rhythm.md]: Are these the only 3 tags? Would it ever make sense to put it on other tags like `body`, `aside`, `footer`, etc.?

Comment (matthewcarleton) [inline source/_patterns/templates/typography/vertical-rhythm.md]: Yes, is there any value in specifying all possible tags, or could we just write "etc"?

File: source/_patterns/templates/typography/vertical-rhythm.hbs (hunk 16 above)
Comment (matthewcarleton) [inline source/_patterns/templates/typography/vertical-rhythm.hbs]: I misspelt that so many times and fixed it :) Good catch!

File: source/_patterns/templates/typography/vertical-rhythm.hbs [hunk 17]
```diff
@@ -0,0 +1,34 @@
+<section class="container-fluid">
```
Comment (andresgalante) [inline source/_patterns/templates/typography/vertical-rhythm.hbs]: Is there any reason why this is a section and not a div?

File: source/scss/_pf-mixins.scss [hunk 18]
```diff
@@ -102,3 +102,18 @@ limitations under the License.
     box-shadow: $top $left $blur $size $color;
   }
 }
+
+// Vertical Rhythm
+//
+// Used in conjunction with .vertical-rhythm to apply size, weight, and margins to type.
```
Comment (andresgalante) [inline source/scss/_pf-mixins.scss]: "to headings" instead of "to type".

File: source/scss/_pf-mixins.scss [hunk 19]
```diff
@@ -102,3 +102,18 @@ limitations under the License.
...
+
+@mixin pf-heading-styles (
+  $weight: null,
+  $size: null,
+  $margin-top: null,
+  $margin-bottom: null) {
+  font-size:$size;
```
Comment (andresgalante) [inline source/scss/_pf-mixins.scss]: human linter

File: source/scss/_pf-type.scss [hunk 20]
```diff
@@ -19,15 +19,73 @@ limitations under the License.
...
+.h4, .pf-vertical-rhythm h4 {
+  @include pf-heading-styles(
+    $weight: $pf-font-weight-semi-bold,
+    $size: $font-size-h4,
+    $margin-top: $pf-headings-margin-top,
+    $margin-bottom: $pf-typography-margin-bottom
+  )
+}
+
+p,ul,ol,dl,blockquote,pre,td,th,textarea {
```
Comment (andresgalante) [inline source/scss/_pf-type.scss]: human linter, space after ,

File: source/scss/_pf-type.scss [hunk 21]
```diff
@@ -19,15 +19,73 @@ limitations under the License.
...
+
+  .pf-vertical-rhythm & {
+    margin-bottom: $pf-typography-margin-bottom;
+    &:last-child {
+      margin-bottom:0; // last item in vertical-rhythm parent has no margin-bottom
```
Comment (andresgalante) [inline source/scss/_pf-type.scss]: human linter

**Comment by andresgalante** [comment]
@matthewcarleton Looks great, I just made a few small comments, mostly linting. Other than that I we can merge this. Thanks a lot!

File: source/_patterns/templates/typography/vertical-rhythm.hbs (hunk 17 above)
Comment (matthewcarleton) [inline source/_patterns/templates/typography/vertical-rhythm.hbs]: I've updated it so the section tag is in the right spot now.

**Pull request merged**